torch/nn/parallel/distributed.py

```diff
@@ -219,20 +254,31 @@ def _dist_broadcast_coalesced(self, tensors, buffer_size):
     def _sync_params(self):
         if len(self.device_ids) > 1:
             # intra-node parameter sync
-            params = [p.data for p in self.module.parameters()]
-            result = broadcast_coalesced(params, self.device_ids, self.broadcast_bucket_size)
-            for tensors, module in zip(result[1:], self._module_copies[1:]):
-                for tensor, param in zip(tensors, module.parameters()):
-                    param.data.set_(tensor)
-
+            for tp in self.param_type_buckets:
+                params = [p.data for p in self.param_type_buckets[tp][0]]
+                result = broadcast_coalesced(params,
+                                             self.device_ids,
+                                             self.broadcast_bucket_size)
+                for idx, tensors in enumerate(result[1:]):
+                    # Just to make it clear
+                    dev_idx = idx + 1
+                    for tensor, param in \
+                            zip(tensors,
+                                self.param_type_buckets[tp][dev_idx]):
+                        param.data.set_(tensor)
```

Comment:
I don't think this change is necessary. `broadcast_coalesced` will handle tensors of different types correctly by itself (the code that we used previously is still ok)

Reply:
Oh, right, the other landed PR has already done this. So many pending PRs on the same file :)